nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/OrderDocument.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume Renard <grenard@nuxeo.com>
+ */
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.impl.DocumentModelListImpl;
+
+@Operation(id = OrderDocument.ID, category = Constants.CAT_DOCUMENT, label = "Order Document", description = "Given a parent document, order the source child before the destination child.")
+public class OrderDocument {
+
+    public static final String ID = "Document.Order";
+
+    @Param(name = "before", required = true)
+    protected DocumentModel before;
+
+    @Context
+    protected CoreSession session;
+
+    @OperationMethod
+    public DocumentModel run(DocumentModel doc) {
+        if (!before.getParentRef().equals(doc.getParentRef())) {
+            throw new NuxeoException("The document can only be ordered within the same folder.");
+        }
+        session.orderBefore(doc.getParentRef(), doc.getName(), before.getName());
+        session.save();
+        return session.getDocument(doc.getRef());
+    }
+
+    @OperationMethod
+    public DocumentModelList run(DocumentModelList docs) {
+        DocumentModelList result = new DocumentModelListImpl();
+        DocumentModel bef = before;
+        while (!docs.isEmpty()) {
```

Comment:
I wonder if you couldn't use a `stream` on `docs`, calling `#run` for each one, something like:
```
        return docs.stream().map(doc -> {
            try {
                return run(doc);
            } catch (OperationException e) {
                throw new NuxeoException(e);
            }
        }).collect(Collectors.toCollection(DocumentModelListImpl::new));
```